build-data-capturing-gradle-samples/capture-test-pts-support/gradle-test-pts-support.gradle

```diff
@@ -1,76 +1,87 @@
 if (!buildScanApi) {
     return
 }
-
-@Field
-final def supportedEngines = [
-    'org.junit.support.testng.engine.TestNGTestEngine' : 'testng',
-    'org.junit.jupiter.engine.JupiterTestEngine' : 'junit-jupiter', 
-    'org.junit.vintage.engine.VintageTestEngine' : 'junit-vintage', 
-    'org.spockframework.runtime.SpockEngine' : 'spock',
-    'net.jqwik.engine.JqwikTestEngine' : 'jqwik',
-    'com.tngtech.archunit.junit.ArchUnitTestEngine' : 'archunit',
-    'co.helmethair.scalatest.ScalatestEngine' : 'scalatest'
-    ]
-
+def capture = new Capture(gradle.rootProject.logger)
 allprojects {
     tasks.withType(Test).configureEach { t ->
         doFirst {
-            if (t.getTestFramework().getClass().getName() == 'org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework') {
-                def engines = testEngines(t) 
-                buildScanApi.value("${t.identityPath}#engines", "${engines}")
-                if (!engines.isEmpty() && engines.stream().allMatch { e -> supportedEngines.containsKey(e) }) {
-                    buildScanApi.value("${t.identityPath}#pts", 'SUPPORTED')
-                } else {
-                    buildScanApi.value("${t.identityPath}#pts", 'ENGINES_NOT_ALL_SUPPORTED')
-                }
+            capture.capturePts(t, buildScanApi)
+        }
+    }
+}
+
+class Capture {
+    final def supportedEngines = [
+        'org.junit.support.testng.engine.TestNGTestEngine' : 'testng',
+        'org.junit.jupiter.engine.JupiterTestEngine' : 'junit-jupiter',
+        'org.junit.vintage.engine.VintageTestEngine' : 'junit-vintage',
+        'org.spockframework.runtime.SpockEngine' : 'spock',
+        'net.jqwik.engine.JqwikTestEngine' : 'jqwik',
+        'com.tngtech.archunit.junit.ArchUnitTestEngine' : 'archunit',
+        'co.helmethair.scalatest.ScalatestEngine' : 'scalatest'
+    ]
+    private Logger logger
+
+    Capture(Logger logger) {
+        this.logger = logger
+    }
+
+    void capturePts(Test t, BuildScanExtension buildScanApi) {
+        if (t.getTestFramework().getClass().getName() == 'org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework') {
+            def engines = testEngines(t)
+            buildScanApi.value("${t.identityPath}#engines", "${engines}")
+            if (!engines.isEmpty() && engines.stream().allMatch { e -> supportedEngines.containsKey(e) }) {
+                buildScanApi.value("${t.identityPath}#pts", 'SUPPORTED')
             } else {
-                buildScanApi.value("${t.identityPath}#pts", 'NO_JUNIT_PLATFORM')
+                buildScanApi.value("${t.identityPath}#pts", 'ENGINES_NOT_ALL_SUPPORTED')
             }
+        } else {
+            buildScanApi.value("${t.identityPath}#pts", 'NO_JUNIT_PLATFORM')
         }
     }
-}
 
-Set<String> testEngines(Test t) {
-    try {
-        Stream<String> engines = t.classpath.files.stream()
+    private Set<String> testEngines(Test t) {
```

Comment:
Default visibility is `public` in Groovy (as well as in Kotlin), here the `private` document which method is the entry point (and ensures private methods are not called within the script, although that could be considered secondary). I would prefer adding the `private` on methods where they're missing (both Kotlin and Groovy), wdyt ?

Reply:
`private` does not change the visibility of the method due to[ this bug in groovy](https://issues.apache.org/jira/browse/GROOVY-1875). I suggest removing `private` since it does nothing.

Reply:
🤯🤯 indeed just tested it and `private` has still no effect, what in the world ?!